gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/tasks/RunTask.groovy

```diff
@@ -23,9 +23,10 @@ class RunTask {
 
     static JavaExec createRunTask(Project project, String taskName) {
         return project.tasks.create(taskName, JavaExec) { t ->
+            t.dependsOn(project.tasks.jar)
```

Comment:
I would expect that the task dependency gets added automatically when one depends on the task output (not the `archivePath`), at least this is how it works in gradle-docker, see https://github.com/palantir/gradle-docker/blob/develop/src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy#L463 . Have you verified the behavior?